docs/pipeline-howto.md

```diff
@@ -1,261 +1,128 @@
+The following sections may need to be customized:
 
->Note: Fork this repository into your own repository for this purpose!).
+- the cluster where your application executed
 
-For setting up your own spark application for the workflow you can start from the `.pipeline.yml` configuration file from `spark-pi-example` and customize it.
-
-The following sections needs to be modified:
+  ```yaml
+  create_cluster:
+    ...
+    cluster_name: "[[your-cluster-name]]"
+  ```
 
 - the command for building your application
 
   ```yaml
   remote_build:
     ...
     original_commands:
-      - mvn clean package -s settings.xml
+      - mvn clean package
   ```
 
 - the Main class of your application
 
   ```yaml
   run:
     ...
-    spark_class: banzaicloud.SparkPi
+    spark_submit_options:
+    ...
+      class: banzaicloud.SparkPi
   ```
 
-- the name of your application
+- the name of your spark application
 
   ```yaml
   run:
     ...
-    spark_app_name: sparkpi
+    spark_submit_configs:
+    ...
+      spark.app.name: sparkpi
   ```
 
 - the application artifact
 
-  This is the relative path to the `jar` of your Spark application. This is the `jar` generated by the [build command](#command-for-building-your-application)
+  This is the relative path to the `jar` of your Spark application. This is the `jar` generated by the [build command](#the-command-for-building-your-application)
 
   ```yaml
   run:
     ...
-    spark_app_source: target/spark-pi-1.0-SNAPSHOT.jar
+    spark_submit_app_args:
+      - target/spark-pi-1.0-SNAPSHOT.jar
   ```
 
--  the application arguments
+- the application arguments
 
   ```yaml
   run:
     ...
-    spark_app_args: 1000
+    spark_submit_app_args:
+      - 1000
   ```
 
 ### Grant access to desired GitHub organizations
 
-Navigate to `http://{control_plane_public_ip}` in your web browser and grant access for the organizations that contain the GitHub repositories that you want to hook into the CI/CD workflow. Then click authorize access.
+Navigate to `http://{control_plane_public_ip}/auth/github/login` in your web browser and grant access for the organizations that contain the GitHub repositories that you want to hook into the CI/CD workflow. Then click authorize access.
+
+If the login and authorization succeeds the user is redirected to the CI/CD UI.
 
 All the services of the Pipeline may take some time to fully initialize, thus the page may not load at first. Please give it some time and retry.
 
 ### Hook repositories to CI/CD flow
 
-Navigate to `http://{control_plane_public_ip}`  - this will bring you to the CI/CD user interface. Select `Repositories` from top left menu. This will list all the repositories that the Pipeline has access to.
+Navigate to `http://{control_plane_public_ip}/`  - this will bring you to the CI/CD user interface. Select `Repositories` from top left menu. This will list all the repositories that the Pipeline has access to.
```

Comment:
Sometimes, it is not enough for me to select the repositories I also need to hit the synchronise button too.

Reply:
This one has beed fixed just yesterday in https://github.com/banzaicloud/pipeline/pull/276